### ISSUE TITLE
Aligning Nanopore reads against a highly similar reference.

### ISSUE THREAD

**Issue by coro1c** [lh3/minimap2#1251]
Hello,

We are doing high-throughput cloning of different gene variants and want to move the QC of our cloning products to Nanopore sequences. I did a trial run with a library of ~400 variants. The variants have large constant parts but differ in two 40 bp stretches.  

When I try to align the Nanopore reads from a flongle (N50: 2.14kb, avg Q-score: 18), I get large alignment errors most likely due to the reference being so similar. I tried several flags (-U, -f to limit kmers in common regions, -G to limit gap length to a few bp, ...). However, my alignment either crashes or gives low quality alignments in the variable regions (in the picture: ~350 and ~1350 bp).
<img width="1523" alt="grafik" src="https://github.com/user-attachments/assets/f4b1e231-1b4a-4e30-97ce-e4ab721d7fc5">
<img width="1506" alt="grafik" src="https://github.com/user-attachments/assets/3c510e26-4092-424e-ba38-a48a5272691b">
<img width="1484" alt="grafik" src="https://github.com/user-attachments/assets/3d9efc73-40c4-41cf-ae2d-aac80b4dfdb3">

Can you maybe give some advice/recommendations, how I can best tackle this alignment problem?

Many thanks in advance.

Best,
Marie

**Comment by lh3** [comment]
Are you using 400 sequences as the reference? What command line are you using exactly? Why do you think the alignment is wrong, instead of the cloning being wrong?

**Comment by coro1c** [comment]
Yes, I use 400 sequences as reference. I am using the default settings of EPI2ME wf-alignment, so I think these are the -x map-ont settings.

I checked for a some reads manually and they actually map back to another reference with some indels. 
So I cannot fully rule out cloning errors but for all reads I checked (~20) I got actually a better fitting reference. I think the largest problem are actually indels due to the low Q score of 18.

Many thanks for your help.

**Comment by lh3** [comment]
Please run minimap2 independently and provide the exact command line. Also what do you mean by "better"? Is the alignment score higher? Eyeballing doesn't really count.
